api/clients/v2/codecs/blob.go

```diff
@@ -27,6 +28,13 @@ type Blob struct {
 
 // DeserializeBlob initializes a Blob from bytes
 func DeserializeBlob(bytes []byte, blobLengthSymbols uint32) (*Blob, error) {
+	if uint32(len(bytes)) > blobLengthSymbols*encoding.BYTES_PER_SYMBOL {
+		return nil, fmt.Errorf(
+			"length (%d bytes) is greater than claimed blob length (%d bytes)",
+			len(bytes),
+			blobLengthSymbols*encoding.BYTES_PER_SYMBOL)
```

Comment:
do we also want to add a check if len(bytes) < blobLengthSymbols*encoding.BYTES_PER_SYMBOL ?

Reply:
Added a comment in [7b9e48e0](https://github.com/Layr-Labs/eigenda/pull/1305/commits/7b9e48e05d0d50dccb7af22a61da6f59e124fc60), addressing why we assert byte length is <=, rather than asserting equality